src/main/scala/cognite/spark/v1/FlexibleDataModelBaseRelation.scala

```diff
@@ -57,61 +57,65 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     })
   }
 
+  private def isReservedAttribute(instanceType: InstanceType, attribute: String) = {
+    val alwaysReservedAttributes: Set[String] = Set("space", "externalId", "_type")
+
+    // type is supported as an alias to _type for edges for legacy reasons.
+    val edgeReservedAttributes: Set[String] = Set("type", "startNode", "endNode")
+    alwaysReservedAttributes.contains(attribute) ||
+    (instanceType == InstanceType.Edge && edgeReservedAttributes.contains(attribute))
+  }
+
   private def extractInstancePropertyValue(key: String, value: InstancePropertyValue): Any =
     FlexibleDataModelRelationUtils.extractInstancePropertyValue(schema.apply(key).dataType, value)
 
-  protected def toInstanceFilter(
+  protected def toFilter(
       instanceType: InstanceType,
+      sparkFilter: Filter,
+      space: Option[String],
+      versionedExternalId: Option[String]
```

Comment:
```suggestion
      viewReference: Option[ViewReference],
```
are those always view attributes and only set/unset together? Would be good to clarify, even better maybe find a more specific name than `viewReference` (or is it clear enough which view it is here?)

Reply:
good point, nicer like this. I think viewReference is fine here, that's what this is